drivers/sensor/htu21d/htu21d.c

```diff
@@ -0,0 +1,175 @@
+/* HTU21D temperature & humidity sensor with i2c
+ *
+ * Copyright (c) 2022 Linumiz
+ * Author: Arunmani <arunmani@linumiz.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT te_htu21d
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <drivers/sensor.h> 
+#include <drivers/i2c.h>
+#include <zephyr/sys/byteorder.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(htu21d, CONFIG_SENSOR_LOG_LEVEL);
+
+#include "htu21d.h"
+
+#define SIXTEEN_BIT		((float)(0xFFFF))    
+#define TEMPERATURE_COEFF_BASE	175.72
+#define TEMPERATURE_COEFF_OFFSET	-46.85
+#define HUMIDITY_COEFF_MUL	125.0
+#define HUMIDITY_COEFF_ADD	-6.0
+#define DECIMAL		10000
+#define MICRO	100
+
+
+struct htu21d_data {
+	const struct device *i2c;
+	uint16_t temp;
+	int16_t humidity;
+};
+
+static int send_device_cmd(struct htu21d_data *drv_data, uint16_t *val, uint8_t cmd)
+{
+	int counter = 0;
+	uint8_t value[3];
+	
+	/*to write register address*/
+	if (i2c_write(drv_data->i2c, &cmd, sizeof(cmd), HTU21D_I2CADDR) != 0) {
+		return -EIO;
+	}
+
+	/*read values*/
+	if (i2c_read(drv_data->i2c, &value, sizeof(value), HTU21D_I2CADDR) != 0) {
+		return -EIO;
+	}
+
+	if(IS_ENABLED(CONFIG_HTU21D_NH_MASTER)) {
+		k_sleep(K_MSEC(MEASURE_WAIT_MS));
+	}
+
+	*val = sys_get_be16(value);
+
+	return 0;
+}
+
+static int htu21d_sample_fetch(const struct device *dev,
+					enum sensor_channel chan)
+{
+	struct htu21d_data *drv_data = dev->data;
+	uint16_t value,value2;
+	drv_data->temp = 0U;
+	drv_data->humidity = 0U;
+	
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		/* Trigger Temperature Measurement*/
+		if (send_device_cmd(drv_data, &value,
+					(IS_ENABLED(CONFIG_HTU21D_NH_MASTER)
+							? HTU21D_READ_TEMP_NH :
+							HTU21D_READ_TEMP_H)) != 0) {
+			return -EIO;
+		}
+
+		drv_data->temp = value;
+		break;
+	case SENSOR_CHAN_HUMIDITY:
+		/* Trigger Humidity Measurement*/
+		if (send_device_cmd(drv_data, &value,
+					(IS_ENABLED(CONFIG_HTU21D_NH_MASTER)
+						? HTU21D_READ_HUMID_NH :
+						HTU21D_READ_HUMID_H)) != 0) {
+			return -EIO;
+		}
+
+		drv_data->humidity = value;
+		break;
+	default:
+		return -ENOTSUP;
+		break;
```

Comment:
unreachable remove it

Reply:
If there is anything other than these two channels thats why i give default

Reply:
any statement after return is unreachable, remove 'break;' statement.. compiler might have thrown warning !!